### DISCUSSION TITLE
Error: Failed to install addin 'Cake.Webpack'

### DISCUSSION THREAD

**Discussion by sinall** [cake-build/cake#3766] [Extension Q&A]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/cake-build/cake/issues) to ensure it has not already been reported

### Cake runner

Cake .NET Tool

### Cake version

2.0.0

### Operating system

Windows

### Operating system architecture

64-Bit

### What are you seeing?

> Error: Failed to install addin 'Cake.Webpack'.

### What is expected?

Install addin 'Cake.Webpack' successfully.

### Steps to Reproduce

build.cake:
```
#addin "Cake.Git"
#addin "Cake.Putty"
#addin "Cake.Webpack"
```

Error:

> Error: Failed to install addin 'Cake.Webpack'.

### Output log

> dotnet cake --verbosity=diagnostic
> Looking for modules...
> No modules found to install.
> Module directory does not exist.
> Analyzing build script...
> Analyzing D:/Documents/source/repos/ShiPanE/branches/master/build.cake...
> Processing build script...
> Installing addins...
> The 'addin' directive is attempting to install the 'Cake.Git' package
> without specifying a package version number.
> More information on this can be found at https://cakebuild.net/docs/writing-builds/reproducible-builds/
> It's not recommended, but you can explicitly override this warning
> by configuring the Skip Package Version Check setting to true
> (i.e. command line parameter "--settings_skippackageversioncheck=true",
> environment variable "CAKE_SETTINGS_SKIPPACKAGEVERSIONCHECK=true",
> read more about configuration at https://cakebuild.net/docs/running-builds/configuration/)
> Resolving assembly 'System.Private.Uri.resources, Version=4.0.6.0, Culture=en-US, PublicKeyToken=b03f5f7f11d50a3a' using runtime installed at 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.21\'...
> Exception occurred while resolving assembly System.Private.Uri.resources: Could not load file or assembly 'System.Private.Uri.resources, Version=4.0.6.0, Culture=en-US, PublicKeyToken=b03f5f7f11d50a3a'. The system cannot find the file specified.
> Trying to resolve assembly System.Private.Uri.resources as 'System.Private.Uri, Version=4.0.6.0, Culture=en-US, PublicKeyToken=b03f5f7f11d50a3a'...
> Assembly System.Private.Uri resolved as 'System.Private.Uri, Version=4.0.6.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' (file location: 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\3.1.21\System.Private.Uri.dll')
>   CACHE https://api.nuget.org/v3/registration5-gz-semver2/cake.git/index.json
> The addin Cake.Git will reference git2-6777db8.dll.
> The addin Cake.Git will reference Cake.Git.dll.
> The addin Cake.Git will reference LibGit2Sharp.dll.
> The 'addin' directive is attempting to install the 'Cake.Putty' package
> without specifying a package version number.
> More information on this can be found at https://cakebuild.net/docs/writing-builds/reproducible-builds/
> It's not recommended, but you can explicitly override this warning
> by configuring the Skip Package Version Check setting to true
> (i.e. command line parameter "--settings_skippackageversioncheck=true",
> environment variable "CAKE_SETTINGS_SKIPPACKAGEVERSIONCHECK=true",
> read more about configuration at https://cakebuild.net/docs/running-builds/configuration/)
>   CACHE https://api.nuget.org/v3/registration5-gz-semver2/cake.putty/index.json
> The addin Cake.Putty will reference Cake.Putty.dll.
> The 'addin' directive is attempting to install the 'Cake.Webpack' package
> without specifying a package version number.
> More information on this can be found at https://cakebuild.net/docs/writing-builds/reproducible-builds/
> It's not recommended, but you can explicitly override this warning
> by configuring the Skip Package Version Check setting to true
> (i.e. command line parameter "--settings_skippackageversioncheck=true",
> environment variable "CAKE_SETTINGS_SKIPPACKAGEVERSIONCHECK=true",
> read more about configuration at https://cakebuild.net/docs/running-builds/configuration/)
>   CACHE https://api.nuget.org/v3/registration5-gz-semver2/cake.webpack/index.json
> Assemblies not found for tfm .NETCoreApp,Version=v3.1 and rid win10-x64.
> No assemblies found after running content resolver.
> Error: Cake.Core.CakeException: Failed to install addin 'Cake.Webpack'.
>    at Cake.Core.Scripting.ScriptProcessor.InstallAddins(IReadOnlyCollection`1 addins, DirectoryPath installPath) in C:\projects\cake\src\Cake.Core\Scripting\ScriptProcessor.cs:line 109
>    at Cake.Core.Scripting.ScriptRunner.Run(IScriptHost host, FilePath scriptPath) in C:\projects\cake\src\Cake.Core\Scripting\ScriptRunner.cs:line 106
>    at Cake.Features.Building.BuildFeature.RunCore(IRemainingArguments arguments, BuildFeatureSettings settings) in C:\projects\cake\src\Cake\Features\Building\BuildFeature.cs:line 99
>    at Cake.Features.Building.BuildFeature.Run(IRemainingArguments arguments, BuildFeatureSettings settings) in C:\projects\cake\src\Cake\Features\Building\BuildFeature.cs:line 49
>    at Cake.Commands.DefaultCommand.Execute(CommandContext context, DefaultCommandSettings settings) in C:\projects\cake\src\Cake\Commands\DefaultCommand.cs:line 73

**Comment by sinall** [comment]
Workaround:
copy `tools\Addins\Cake.Webpack.0.3.0\lib\net461` to `tools\Addins\Cake.Webpack.0.3.0\lib\net6.0`

**Comment by sinall** [comment]
Another problem for 2.0.0 Cake.Webpack (or cakebuild) is that the output format:

> ========================================
> BuildAdmin
> ========================================
> Browserslist: caniuse-lite is outdated. Please run next command `npm update`
>                                                                             Hash: a79e6471ae3387e9db06
>                                                                                                       Version: webpack 3.12.0
>      Time: 5774ms
>                                                                   Asset       Size  Chunks                    Chunk Names
>                                            admin/app.js     382 kB       0  [emitted]  [big]  admin/app.js

The format goes malformed.

**Comment by devlead** [comment]
Cake.Webpack is a community maintained addin, please raise an issue in its repository
https://github.com/cake-contrib/Cake.Webpack/issues/new

A quick look at the addin it looks like it targets Cake version 0.33.0 and .NET Framework v4.6.1.

So for it to work as intended with Cake 2.0 it would need to be updated to use Cake.Core 2.0 and target netcoreapp3.1, net5.0, and net6.0. If you have the time and opportunity I'm sure a PR addressing this to assist the maintainer would be welcome.
